### ISSUE TITLE
Custom API request

### ISSUE THREAD

**Issue by rht** [zulip/zulip-terminal#227]
This is a feature request for ZT to have option for user to input custom API request to the server (this could either be specified in cfg or in a colon-prefixed command input analogous to vim's command-line mode). This is particularly useful for debug purpose, but additionally could be a basis for adding another feature: a custom shortcut at the left-sidebar that automatically display messages from a search result of a certain keyword. This feature (the custom filter shortcut) has long existed in gmail, in the form of filters/labels.

**Comment by amanagr** [comment]
@rht I feel this feature could be easily implemented. The only issue is creating standards for taking input for these commands, which can be based on the parameters of the api calls but then we will need to document them properly.

Also, knowing which type of filters people generally use can be a big plus in designing them. Perhaps you can list some of the filters you desire to use here.

**Comment by rht** [comment]
I will have to draw from the filters I have used in gmail. Most of them are in the form of labeling emails sent to mailing lists (which is already solved by streams), and keyword matching for a theme (e.g. "nlp" OR "deep learning" OR "reinforcement learning", the kind that tends to be scattered across topics or streams, but this is also helpful to group a set of topics within a stream, just like how I can focus on the subset of a content of a mailing list via the keywords filter). For the latter, there is already the alert word feature, but it is very limited in that it functions as an on-the-spot notification rather than a log that can be revisited.

-- this is 1 data point

**Comment by neiljp** [comment]
@rht I think we're much closer to a situation where this is reasonable or feasible. To clarify, you're essentially talking about user-customizable 'bookmark' buttons? These could be specified in some local configuration, saved from a particular search 'result', or another method. Does that fit what you had in mind? If so, I think we should update the title to reflect this.

**Comment by rht** [comment]
> To clarify, you're essentially talking about user-customizable 'bookmark' buttons?

Nah, it was just an example application, which can be put into another issue. What I want is raw API request to the server.
